### PR TITLE
Removed course-centric event tracking

### PR DESCRIPTION
**Pull request by clintonb**
Prior to this change course ID was treated special compared to other
event properties. This change treats the course ID as any other
property. Given that we have no course-centric pages, course ID will
almost never be included in the event properties going forward.

LEARNER-278

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce @edx/helio please review

File: ecommerce/extensions/analytics/utils.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def audit_log(name, **kwargs):
     logger.info(message)
 
 
-def prepare_analytics_data(user, segment_key, course_id=None):
+def prepare_analytics_data(user, segment_key):
     """ Helper function for preparing necessary data for analytics.
```
Comment (vkaracic) [inline ecommerce/extensions/analytics/utils.py:91]: Please remove ``course_id`` from the docstring as well.

File: ecommerce/static/js/views/analytics_view.js [hunk 2]
```diff
@@ -102,31 +89,16 @@ define([
...
-             * to Segment.
-             *
-             * @param eventType String event type.
-             */
+          /**
+           * Catch 'segment:track' events and create events and send
+           * to Segment.
+           *
+           * @param eventType String event type.
+           * @param properties Event properties
```
Comment (vkaracic) [inline ecommerce/static/js/views/analytics_view.js]: Nit: dot at the end.

**Review by vkaracic — APPROVED** [review]

File: ecommerce/static/js/views/analytics_view.js (hunk 2 above)
Comment (clintonb) [reply ecommerce/static/js/views/analytics_view.js]: Not a sentence. No need for a period.

**Pull request merged**